packages/itwin/tree-widget/src/test/trees/category-tree/CategoriesTreeComponent.test.tsx

```diff
@@ -27,31 +40,215 @@ describe("<CategoriesTreeComponent />", () => {
+        sinon.stub(categoryVisibilityHandler, "useCategories").returns([]);
+        const categoryTreeSpy = sinon.stub(categoryTree, "CategoryTree").returns(<></>);
+        const spy = sinon.stub().returns(<></>);
+        render(
+          <CategoriesTreeComponent
+            headerButtons={[spy]}
+          />
+        );
+        await waitFor(() => expect(categoryTreeSpy).to.be.called);
+        categoryTreeSpy.args[0][0].onFilterApplied!({ getNodes: async () => [] } as unknown as IFilteredPresentationTreeDataProvider, 0);
+        await waitFor(() => expect(spy).to.be.calledWith(sinon.match((props: CategoriesTreeHeaderButtonProps) => props.filteredCategories !== undefined)));
+      });
+
+      it("filteredCategories return empty filteredCategories array when nodes are not PresentationTreeNodeItem", async () => {
+        sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+        sinon.stub(UiFramework, "getIModelConnection").returns(iModel);
+        sinon.stub(categoryVisibilityHandler, "useCategories").returns([]);
+        const categoryTreeSpy = sinon.stub(categoryTree, "CategoryTree").returns(<></>);
+        const spy = sinon.stub().returns(<></>);
+        render(
+          <CategoriesTreeComponent
+            headerButtons={[spy]}
+          />
+        );
+        await waitFor(() => expect(categoryTreeSpy).to.be.called);
+        categoryTreeSpy.args[0][0].onFilterApplied!({ getNodes: async () => [{ id: "nodeId1" }, { id: "nodeId2" }] } as unknown as IFilteredPresentationTreeDataProvider, 0);
+        await waitFor(() => expect(spy).to.be.calledWith(sinon.match((props: CategoriesTreeHeaderButtonProps) => props.filteredCategories !== undefined && props.filteredCategories.length === 0)));
+      });
+
+      it("filteredCategories returns filteredCategories when nodes are PresentationTreeNodeItem", async () => {
+        sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+        sinon.stub(UiFramework, "getIModelConnection").returns(iModel);
+        sinon.stub(categoryVisibilityHandler, "useCategories").returns([]);
+        const categoryTreeSpy = sinon.stub(categoryTree, "CategoryTree").returns(<></>);
+        const spy = sinon.stub().returns(<></>);
+        render(
+          <CategoriesTreeComponent
+            headerButtons={[spy]}
+          />
+        );
+        await waitFor(() => expect(categoryTreeSpy).to.be.called);
+        categoryTreeSpy.args[0][0].onFilterApplied!({ getNodes: async () => categoryNodes } as unknown as IFilteredPresentationTreeDataProvider, 0);
+        await waitFor(() => expect(spy).to.be.calledWith(sinon.match((props: CategoriesTreeHeaderButtonProps) => (props.filteredCategories !== undefined && props.filteredCategories.length === 2))));
+      });
+
+      it("filteredCategories returns filteredCategories with filteredSubcategories when nodes are PresentationTreeNodeItem", async () => {
```

Comment:
Looking at test setup this test does same thing as previous one. I think they can be combined into one test.

Reply:
fixed?